src/core/fbcontainer.cpp

```diff
@@ -36,67 +34,68 @@ EMGMResponse checkForActionEquivalentState(const CFunctionBlock &paFB, const EMG
   return EMGMResponse::InvalidState;
 }
 
-CFBContainer::CFBContainer(CStringDictionary::TStringId paContainerName, CFBContainer &paParent) :
-    mContainerName(paContainerName), mParent(paParent) {
-}
-
-CFBContainer::CFBContainer(CStringDictionary::TStringId paContainerName, CFBContainer &paParent, size_t paNumFBs) :
-    mContainerName(paContainerName), mParent(paParent) {
-  mFunctionBlocks.reserve(paNumFBs);
+CFBContainer::CFBContainer(CStringDictionary::TStringId paContInstanceName, CFBContainer &paParent) :
+    mContInstanceName(paContInstanceName), mParent(paParent) {
 }
 
 CFBContainer::~CFBContainer() {
-  for (TFunctionBlockList::iterator itRunner(mFunctionBlocks.begin()); itRunner != mFunctionBlocks.end(); ++itRunner) {
-    CTypeLib::deleteFB(*itRunner);
-  }
-  mFunctionBlocks.clear();
-
-  for (TFBContainerList::iterator itRunner(mSubContainers.begin()); itRunner != mSubContainers.end(); ++itRunner) {
+  for (TFBContainerList::iterator itRunner(mChildren.begin()); itRunner != mChildren.end(); ++itRunner) {
     delete (*itRunner);
   }
-  mSubContainers.clear();
+  mChildren.clear();
 }
 
-EMGMResponse CFBContainer::addFB(CFunctionBlock* paFuncBlock){
-  EMGMResponse eRetVal = EMGMResponse::InvalidObject;
+EMGMResponse CFBContainer::addFB(CFunctionBlock* paFuncBlock) {
+  EMGMResponse retVal = EMGMResponse::InvalidObject;
   if(nullptr != paFuncBlock){
-    mFunctionBlocks.insert(getFBIterator(paFuncBlock->getInstanceNameId()), paFuncBlock);
-    eRetVal = EMGMResponse::Ready;
+    mChildren.insert(getChildrenIterator(paFuncBlock->getInstanceNameId()), paFuncBlock);
+    retVal = EMGMResponse::Ready;
+  }
+  return retVal;
+}
+
+EMGMResponse CFBContainer::addDirectlyInstantiatedFB(CFunctionBlock* paFuncBlock) {
+  EMGMResponse retVal = EMGMResponse::InvalidObject;
+  if(paFuncBlock != nullptr) {
+    if(paFuncBlock->initialize()) {
```

Comment:
Should we not better do this in a (generated) `initialize()` of the outer FB?

Reply:
I thought it would be better to keep it at some central place.

Reply:
The current implementation in https://github.com/eclipse-4diac/4diac-ide/pull/332 generates these calls into the constructor without checking the return value. Maybe we could call this from a generated `initialize()` instead and also check for the return value there?